app/serializers/api/v1/stockit_item_serializer.rb

```diff
@@ -31,6 +32,32 @@ def include_order?
 
     alias_method :include_designation_id?, :include_order?
 
+    def include_in_hand_quantity?
+      @options[:include_in_hand_quantity]
+    end
+
+    def include_added_quantity?
+      @options[:include_added_quantity]
+    end
+
+    def in_hand_quantity
+      sql = "select sum(quantity) from packages_inventories where package_id = #{object.id}"
```

Comment:
do we know what the performance impact of this is?

Reply:
@steveyken , There won't be much, because it's not called for all items, `in_hand_quantity` it's called only for 25 items at a time, while searching items for adding it to a box. 
So I don't think this will make any performance impact. 
I have used this in the serializer, 
```  def include_in_hand_quantity?
      @options[:include_in_hand_quantity]
    end```

It is false otherwise, but becomes true only when we're searching for items to add on a box or a pallet. 

Using this param I pass this option to the serializer 
`include_in_hand_quantity: !!params["filter_box_pallet"],`
and this param is sent from the client only while searching for items to add on a box or a pallet.

Thanks

Reply:
Thanks! A very comprehensive answer 💯